actionview/lib/action_view/renderer/partial_renderer.rb

```diff
@@ -456,12 +456,17 @@ def collection_without_template
           index = partial_iteration.index
           path, as, counter, iteration = collection_data[index]
 
+          if layout = @options[:layout]
+            layout = (cache[layout] ||= find_template(layout, keys + [as, counter, iteration]))
```

Comment:
I'm not sure. If this is a problem, then the same problem exists where we're caching the templates for objects in the collection based on the `path`. e.g., https://github.com/stevenharman/rails/blob/5e27a9963dc91baa5ac9ef378e89d7164e895b36/actionview/lib/action_view/renderer/partial_renderer.rb#L467

In the case of the `layout` option, only a single value can be given for that option, so it won't change based on the objects in the collection. If those values really are needed in the cache key - for both the layout and object template, then we can add them. But there's also some caching happening further down in the resolvers, so maybe it's totally unneeded here? https://github.com/stevenharman/rails/blob/5e27a9963dc91baa5ac9ef378e89d7164e895b36/actionview/lib/action_view/template/resolver.rb#L145-L149

Thoughts?